builder/azure/arm/builder_acc_test.go

```diff
@@ -293,63 +315,76 @@ func checkTemporaryGroupDeleted(t *testing.T, b *Builder) {
+	}
+	osState := compute.OperatingSystemStateTypesGeneralized
+	if params.specialized {
+		osState = compute.OperatingSystemStateTypesSpecialized
+	}
+	osArch := compute.ArchitectureArm64
+	if params.isX64 {
+		osArch = compute.ArchitectureX64
+	}
+	hyperVGeneration := compute.HyperVGenerationV1
+	if params.useGenTwoVM {
+		hyperVGeneration = compute.HyperVGenerationV2
+	}
+	location := "southcentralus"
+	future, err := azureClient.GalleryImagesClient.CreateOrUpdate(context.TODO(), "packer-acceptance-test", "acctestgallery", params.galleryImageName, compute.GalleryImage{
+		GalleryImageProperties: &compute.GalleryImageProperties{
+			OsType:           osType,
+			OsState:          osState,
+			Architecture:     osArch,
+			HyperVGeneration: hyperVGeneration,
+			Identifier: &compute.GalleryImageIdentifier{
+				Publisher: &params.imagePublisher,
+				Offer:     &params.imageOffer,
+				Sku:       &params.imageSku,
+			},
+		},
+		Location: &location,
+	})
 
-	spnCloud, spnKeyVault, err := b.getServicePrincipalTokens(ui.Say)
 	if err != nil {
-		t.Fatalf("failed getting azure tokens: %s", err)
+		t.Fatalf("failed to create Gallery %s: %s", params.galleryImageName, err)
 	}
-
-	azureClient, err := NewAzureClient(
-		b.config.ClientConfig.SubscriptionID,
-		b.config.SharedGalleryDestination.SigDestinationSubscription,
-		b.config.ResourceGroupName,
-		b.config.StorageAccount,
-		b.config.ClientConfig.CloudEnvironment(),
-		b.config.SharedGalleryTimeout,
-		b.config.PollingDurationTimeout,
-		spnCloud,
-		spnKeyVault)
-
+	err = future.WaitForCompletionRef(context.TODO(), azureClient.GalleryImagesClient.Client)
 	if err != nil {
-		t.Fatalf("failed to create azure client: %s", err)
+		t.Fatalf("failed to create Gallery %s: %s", params.galleryImageName, err)
 	}
+}
 
-	// validate temporary os blob was deleted
-	blob := azureClient.BlobStorageClient.GetContainerReference("images").GetBlobReference(b.config.tmpOSDiskName)
-	_, err = blob.BreakLease(nil)
-	if err != nil && !strings.Contains(err.Error(), "BlobNotFound") {
-		t.Fatalf("failed validating deletion of unmanaged vhd: %s", err)
+func deleteSharedImageGalleryDefinition(t *testing.T, azureClient AzureClient, galleryImageName string) {
+	versionFuture, err := azureClient.GalleryImageVersionsClient.Delete(context.TODO(), "packer-acceptance-test", "acctestgallery", galleryImageName, "1.0.0")
+	if err != nil {
+		t.Fatalf("failed to delete Gallery %s: %s", galleryImageName, err)
 	}
-
-	// Validate resource group has been deleted
-	_, err = azureClient.GroupsClient.Get(context.Background(), b.config.tmpResourceGroupName)
-	if err == nil || !resourceNotFound(err) {
-		t.Fatalf("failed validating resource group deletion: %s", err)
+	err = versionFuture.WaitForCompletionRef(context.TODO(), azureClient.GalleryImageVersionsClient.Client)
+	if err != nil {
+		t.Fatalf("failed to delete Gallery %s: %s", galleryImageName, err)
+	}
+	// WaitForCompletionRef is unreliable
+	time.Sleep(5000)
```

Comment:
This will likely work but is indeed a bit flaky, I presume we could periodically check for the status of the SIG with like a GET or some other method, and give it a more generous timeout like a minute? Still flaky if the platform is slower than usual for some reason, but likely a bit less than a 5s wait.

Reply:
I resolved this by using a retry from the packer-plugin-sdk